### ISSUE TITLE
[Question] Dynamic load mutiple lora model?

### ISSUE THREAD

**Issue by big-thousand** [oobabooga/text-generation-webui#643]
**Description**

I notice that it can use lora model now.
I want to dynamic load and unload lora model like AUTOMATIC1111/stable-diffusion-webui
How can I do like this

**Comment by Arargd** [comment]
@oobabooga Not sure how useful this is, but I can merge the lora into the model and then load a lora on top of that. No idea if it lets you merge a second and let you use a third, but I know that it works for at least having 2 at the same time. (I only have 2 distinct loras to play with on me right now, so someone let me know if you can do more!)

```
  key_list = [key for key, _ in model.base_model.model.named_modules() if "lora" not in key]
    for key in key_list:
        parent, target, target_name = model.base_model._get_submodules(key)
        if isinstance(target, peft.tuners.lora.Linear):
            bias = target.bias is not None
            new_module = torch.nn.Linear(target.in_features, target.out_features, bias=bias)
            model.base_model._replace_module(parent, target_name, new_module, target)
```

Taken from a pull on the peft github https://github.com/huggingface/peft/pull/227/commits/4289364941ffab7e964f26b2d0cb45f29a92ecfb

**Comment by oobabooga** [comment]
@Arargd is the effect of that the same as taking the average of the two LoRAs? How is this interpreted?

I was thinking of the possibility of literally adding the LoRAs and dividing the parameters by 2 as in a stable diffusion checkpoint merge, but that probably doesn't make sense because the LoRAs can have different numbers of parameters.

**Comment by Arargd** [comment]
> @Arargd is the effect of that the same as taking the average of the two LoRAs? How is this interpreted?
> 
> I was thinking of the possibility of literally adding the LoRAs and dividing the parameters by 2 as in a stable diffusion checkpoint merge, but that probably doesn't make sense because the LoRAs can have different numbers of parameters.

Woops, ignore my other comment. I must've messed with my seed or something, it seems like it had no affect on the output.

**Comment by oobabooga** [comment]
That was still an interesting idea to try.

**Comment by Arargd** [comment]
> That was still an interesting idea to try.

Yeah I'll add this to avoid confusion:

Currently looking at https://github.com/oobabooga/text-generation-webui/issues/643#issuecomment-1489530789 to try and merge the lora to the model itself, from there who knows. Thought I had it working, but I messed up testing somewhere or another.
